ui/src/components/EdgePins.tsx

```diff
@@ -0,0 +1,154 @@
+
+  return L.latLng(a + lambda * (c - a), b + lambda * (d - b));
+}
+
+function angleFromCoordinate(lat1, long1, lat2, long2) {
+  const dLon = long2 - long1;
+
+  const y = Math.sin(dLon) * Math.cos(lat2);
+  const x =
+    Math.cos(lat1) * Math.sin(lat2) -
+    Math.sin(lat1) * Math.cos(lat2) * Math.cos(dLon);
+
+  const brng = (Math.atan2(y, x) * 180) / Math.PI;
+
+  return brng - 80;
+}
+
+function getPOIInfo(pin: LatLng, corners: LatLng[], currPosition: LatLng, edgePinsInfo: { pos: LatLng; angle: number; edgeType: EdgeType }[]) {
+  // iterate through the 4 sides of our current map view and find the point
+  // of intersection between the line from the center of the map view to pin
+  // and the side we are currently looking at
+  for (let i = 0; i < 4; i++) {
+    const j = i + 1 < 4 ? i + 1 : 0;
+    // check if we have an intersection
+    if (
+      intersects(
+        corners[i].lat,
+        corners[i].lng,
+        corners[j].lat,
+        corners[j].lng,
+        currPosition.lat,
+        currPosition.lng,
+        pin.lat,
+        pin.lng
+      )
+    ) {
+      // get the point of intersection
+      const position = getPointOfIntersection(
+        corners[i].lat,
+        corners[i].lng,
+        corners[j].lat,
+        corners[j].lng,
+        currPosition.lat,
+        currPosition.lng,
+        pin.lat,
+        pin.lng
+      );
+      // determine the angle of rotation for the arrow pin
+      const angle = angleFromCoordinate(
+        (currPosition.lat * Math.PI) / 180,
+        (currPosition.lng * Math.PI) / 180,
+        (pin.lat * Math.PI) / 180,
+        (pin.lng * Math.PI) / 180
+      );
+      const pinInfo = {
+        pos: position,
+        angle: angle,
+        edgeType: i,
+      };
+      edgePinsInfo.push(pinInfo);
+      break;
+    }
+  }
+}
+
+export interface EdgePinsProps {
+  stories: { lat: number; lng: number }[];
+  currPosition: LatLng;
+  mapBounds: LatLngBounds;
+}
+
+export function EdgePins({
+  stories,
+  currPosition,
+  mapBounds,
+}: EdgePinsProps): JSX.Element {
+  // determine if the current view has markers
+  if (!mapBounds || !currPosition) return null;
+
+  const pinPositions = stories.map((story) => {
+    return { lat: story.lat, lng: story.lng };
+  });
+
+  const isVisible = !pinPositions.some((pos) => mapBounds.contains(L.latLng(pos.lat, pos.lng)));
+
+  // do not display the edge pins if there are markers in view
+  if (!isVisible) return null;
+
+  // determine the distance from the center to each of the markers
+  const distanceToPos: { [distance: number]: LatLng } = {};
+  pinPositions.forEach((pos) => {
+    const posLatLng = L.latLng(pos.lat, pos.lng);
+    distanceToPos[currPosition.distanceTo(posLatLng)] = posLatLng;
+  });
+
+  //sort the distances and take the first one
+  const distances = Object.keys(distanceToPos).sort().slice(0, 1);
+
+  // get the closest three pins
+  const closestPins: LatLng[] = distances.map((distance) => {
```

Comment:
Isn't distances is a single element array because of `slice(0, 1)`?

Reply:
Yeah this is because design changed it from the 3 closest to the closest one, so I made it so we could have more than one, but now it is just one. Not sure if I should leave it in case we want more in the future?